share/spice/thesaurus/thesaurus.js

```diff
@@ -2,83 +2,90 @@
+        var script = $('[src*="/js/spice/thesaurus/"]')[0];
+        var source = $(script).attr("src");
+        var query = source.match(/thesaurus\/([^\/]+)/)[1];
+        query = decodeURIComponent(query);
+        
+        // Search the results for the word type (noun, adjective etc) with the most entries
+        var wordTypeWithMostEntries;
+        var largestWordTypeEntryTotal = 0;
+        
+        Object.keys(api_result).forEach(function(wordType) { // wordType = ajective, noun etc...
+            var wordTypeEntryTotal = 0;
 
-        var shorthand = {
-            "synonyms"  : "syn",
-            "synonym"   : "syn",
-            "antonyms"  : "ant",
-            "antonym"   : "ant",
-            "related"   : "rel",
-            "similar"   : "sim",
-            "thesaurus" : "syn"
-        };
+            Object.keys(api_result[wordType]).forEach(function(relationshipType) { // relationshipType = synonym, antonym etc...
+                wordTypeEntryTotal += Object.keys(api_result[wordType][relationshipType]).length;
+            });
 
-        var headers = {
-            "syn" : "Synonyms of ",
-            "ant" : "Antonyms of ",
-            "rel" : "Related to ",
-            "sim" : "Similar to "
-        };
-
-        // Check if the mode exists.
-        var how_many = 0;
-        for(var i in api_result) {
-            if(api_result.hasOwnProperty(i) && (shorthand[mode] in api_result[i])) {
-                how_many += 1;
+            if (wordTypeEntryTotal > largestWordTypeEntryTotal) {
+                largestWordTypeEntryTotal = wordTypeEntryTotal;
+                wordTypeWithMostEntries = wordType;
             }
-        }
-        if(how_many === 0) {
-            return Spice.failed('thesaurus');
+        });
+        
+        if (!wordTypeWithMostEntries) {
+            return Spice.failed("thesaurus");    
         }
 
-        api_result.mode = shorthand[mode];
-
-        // Create the plugin.
         Spice.add({
             id: 'thesaurus',
             name: 'Thesaurus',
-            data:  api_result,
-            normalize: function(item){
-                var res = {
-                    headerText: headers[item.mode] || 'Thesaurus: ',
-                    query: query,
-                    results: []
-                };
-
-                for(var parts_of_speech in item) {
-                    if(item.hasOwnProperty(parts_of_speech) && item[parts_of_speech][item.mode]) {
-                        res.results.push({
-                            heading : parts_of_speech.charAt(0).toUpperCase() + parts_of_speech.slice(1),
-                            words   : item[parts_of_speech][item.mode].splice(0, 10).join(", ")
-                        });
-                    }
+            data:  api_result[wordTypeWithMostEntries],
+            templates: {
+                group: 'list',
+                options:{
+                    content: 'record', 
+                    rowHighlight : true,
```

Comment:
Wow, I didn't realize this still works. We opted to remove it from all IA's because the design team didn't want it any more. Let's remove that so it looks like all our other "List" IAs